scripts/backupmanager.sh

```diff
@@ -0,0 +1,336 @@
+        pp --error "> Exiting...\n"
+        exit 1
+    elif [ -z "${SAVE_DIR}" ]; then
+        pp --error "> GAME_SAVE_PATH environment variable not set.\n"
+        pp --error "> Exiting...\n"
+        exit 1
+    fi
+
+    # If vars are set, check if the directories exist
+    if [ ! -d "${BACKUP_DIR}" ]; then
+        pp --error "[ERROR] Backup directory '${BACKUP_DIR}' does not exist.\n"
+        exit 1
+    elif [ ! -d "${SAVE_DIR}" ]; then
+        pp --error "[ERROR] Game save directory '${SAVE_DIR}' does not exist.\n"
+        exit 1
+    fi
+}
+
+
+### Backup Functions
+
+function create_backup() {
+
+    pp --success ">>> Creating backup...\n\n"
+
+
+    if [ -z "${GAME_PATH}" ]; then
+        pp --error "> RESTORE_FILE_PATH environment variable not set. Exiting...\n"
+        exit 1
+    fi
+    
+    DATE=$(date +%Y%m%d_%H%M%S)
+    TIME=$(date +%H-%M-%S)
+
+    #Send message to gameserver via RCON if enabled
+    rc "broadcast ${TIME}-Backup_in_progress" "> Broadcasting server shutdown via RCON...\n"
+
+    # Create backup dir if it doesn't exist
+    mkdir -p "${BACKUP_DIR}"
+
+    #Send message to gameserver via RCON if enabled
+    rc 'broadcast Saving world...' "> Sending message to gameserver via RCON..."
+    rc 'save'
+    rc 'broadcast Done!' "> Broadcast via RCON complete!"
+
+
+    # Create backup
+    tar cfz "${BACKUP_DIR}/saved-${DATE}.tar.gz" -C "${GAME_PATH}" Saved/
+    pp --success ">>> Backup created successfully!\n"
+
+
+}
+
+function list_backups() {
+    local num_backup_entries=${1:-""}
+
+    pp --success ">>> Listing backups:\n\n"
+
+    # Check num_backup_entries argument
+    # if argument exists and is not a positive integer, print usage message and exit
+    if [[ -n "$num_backup_entries" && ! "$num_backup_entries" =~ ^[0-9]+$ ]]; then
+        pp --error "[ERROR] Invalid argument. Please provide a positive integer.\n"
+        exit 1
+    fi
+
+    # Check if backup directory exists
+    if [ ! -d "$BACKUP_DIR" ]; then
+        pp --error "[ERROR] Backup directory $BACKUP_DIR does not exist.\n"
+        exit 1
+    fi
+
+    # Check if there are any files in the backup directory
+    if [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+        pp --info "No backups in the backup directory $BACKUP_DIR.\n"
+        exit 0
+    fi
+
+    # get file list
+    files=$(find "$BACKUP_DIR" -maxdepth 1 -type f -name "*.tar.gz" | sort -r)
+    total_file_count=$(echo "$files" | wc -l)
+
+    if [ -z "$num_backup_entries" ]; then
+        # if num_backup_entries is not set, use all files
+        file_list=$files
+    else
+        # if num_backup_entries is set, use the first num_backup_entries files
+        file_list=$(echo "$files" | head -n "$num_backup_entries")
+    fi
+
+    # print file list (currently using date from file name but can use creation date from 'stat')
+    for file in $file_list; do
```

Comment:
Why do this so complicated?
Why parse something?
ls and the filename is all you need, aint it?